WIP: Notifications: Add Simperium library

dmsnell opened the pull request:
This patch introduces the Simperium library in preparation to allow
sourcing the notification data from Simperium instead of via polling
the WordPress.com API.

In this patch we're limiting our work to adding the new dependency
and getting it properly configured.

---

When building and running I get errors from `webpack` in the build
output and also in the browser…

```
Module not found: Error: Can't resolve 'https' in '…'
Field 'browser' doesn't contain a valid alias configuration
```

I believe it's related to how `simperium` can be used in `node` and
in `browser` environments.

dmsnell committed:
WIP: Notifications: Add Simperium library

This patch introduces the Simperium library in preparation to allow
sourcing the notification data from Simperium instead of via polling
the WordPress.com API.

In this patch we're limiting our work to adding the new dependency
and getting it properly configured.

blowery commented on apps/notifications/package-lock.json:
```diff
@@ -0,0 +1,153 @@
+{
```
apps no longer have their own package locks. You should be able to drop this file entirely. After you add the dependency to notifications/package.json, run `npm i` in the root until you get a stable package-lock.json in the root.

dmsnell replied:
thanks @blowery - I updated but it didn't seem to address the the issue at hand, not sure if it was supposed to.

dmsnell committed:
Remove project-local package-lock.json

dmsnell committed:
Remove `node` exclusion from `webpack` configs

This introduces a change that allows `webpack` to include polyfills for
`node` libraries that otherwise wouldn't run in the browser.

As work progresses we might replace this with updates to `node-simperium`
that don't depend on the `node` libraries. In the meantime this measure
will allow us to proceed in developing changes to the notifications panel
without having to address this deeper issue.

dmsnell committed:
Complete functionality of notes panel in Simperium client

dmsnell committed:
How did that get in there?

dmsnell committed:
Update to "universal" branch of `node-simperium`

dmsnell committed:
Move dynamic-loading of `simperium` library deeper into notifications panel

dmsnell committed:
Add chunk name

dmsnell committed:
Rely on the newly-published node-simperium library

dmsnell committed:
update package-lock.json
